jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1100,29 +1220,55 @@ public MethodHandle findStaticSetter(Class<?> clazz, String fieldName, Class<?>
 			return handle;
 		}
 		
+		/*[IF Java14]*/
 		/**
 		 * Create a lookup on the request class.  The resulting lookup will have no more 
 		 * access privileges than the original.
 		 * 
 		 * @param lookupClass - the class to create the lookup on
 		 * @return a new MethodHandles.Lookup object
+		 * @throws NullPointerException - if lookupClass is null
+		 * @throws IllegalArgumentException - if the requested Class is a primitive type or an array class
 		 */
-		public MethodHandles.Lookup in(Class<?> lookupClass){
-			lookupClass.getClass();	// implicit null check
+		public MethodHandles.Lookup in(Class<?> lookupClass) throws NullPointerException, IllegalArgumentException {
+		/*[ELSE]*/
+		/**
+		 * Create a lookup on the request class.  The resulting lookup will have no more 
+		 * access privileges than the original.
+		 * 
+		 * @param lookupClass - the class to create the lookup on
+		 * @return a new MethodHandles.Lookup object
+		 */
+		public MethodHandles.Lookup in(Class<?> lookupClass) {
+		/*[ENDIF] Java14 */
+			if (Objects.isNull(lookupClass)) {
+				/*[MSG "K0682", "The requested lookup class must not be null"]*/
+				throw new NullPointerException(com.ibm.oti.util.Msg.getString("K0682")); //$NON-NLS-1$
+			}
+			
+			/*[IF Java14]*/
+			if ((lookupClass == void.class) || lookupClass.isPrimitive() || lookupClass.isArray()) {
```

Comment:
What's the expected behaviour of these cases pre-java14?  We may be able to enable this everywhere

Reply:
there is no other check on the requested lookup class except the null-check according to the Spec at Java 11, 12 and 13 at 
http://cr.openjdk.java.net/~iris/se/11/build/latest/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#in(java.lang.Class)
http://cr.openjdk.java.net/~iris/se/12/build/latest/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#in(java.lang.Class)
http://cr.openjdk.java.net/~iris/se/13/build/latest/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#in(java.lang.Class)
```
public MethodHandles.Lookup in(Class<?> requestedLookupClass)
...
Throws:
    NullPointerException - if the argument is null
```
against the Java 14 Spec
```
public MethodHandles.Lookup in(Class<?> requestedLookupClass)
...
Throws:
  --> IllegalArgumentException - if requestedLookupClass is a primitive type or void or array class
    NullPointerException - if the argument is null
```
And such check is not required in other APIs in Java 14 except `Lookup::in`.
So it should never hurt to enable the check here (only internal tests need to be updated to deal with the check)

Reply:
I used `jshell` to test an RI JDK 10 as it was conveniently available and it doesn't have the check so we can't enable everywhere:

```
$ ../../jdk-10.0.2+13/bin/jshell -J-showversion
openjdk version "10.0.2-adoptopenjdk" 2018-07-17
OpenJDK Runtime Environment (build 10.0.2-adoptopenjdk+13)
OpenJDK 64-Bit Server VM (build 10.0.2-adoptopenjdk+13, mixed mode)
|  Welcome to JShell -- Version 10.0.2-adoptopenjdk
|  For an introduction type: /help intro

jshell> import java.lang.invoke.*

jshell> MethodHandles.Lookup l = MethodHandles.lookup();
l ==> 

jshell> l.in(void.class)
$3 ==> void/public

jshell> l.in(int.class)
$4 ==> int/public

jshell> l.in(int[].class)
$5 ==> [I/public

```

Reply:
I double-checked on RI/Hotspot at JDK 11, 12, 13 and 14 as follows and it seem only RI JDK14 does the type check.
1. JDK11
```
# jdk_11.0.6_10_hotspot/bin/jshell -J-showversion
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
Mar 02, 2020 9:54:53 PM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
|  Welcome to JShell -- Version 11.0.6
|  For an introduction type: /help intro

jshell> import java.lang.invoke.*

jshell> MethodHandles.Lookup l = MethodHandles.lookup();
l ==>
jshell> l.in(void.class)
$3 ==> void/public

jshell> l.in(int.class)
$4 ==> int/public

jshell> l.in(int[].class)
$5 ==> [I/public
```
2. JDK12
```
# jdk_12.0.2_10_hotspot/bin/jshell -J-showversion
openjdk version "12.0.2" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.2+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 12.0.2+10, mixed mode, sharing)
|  Welcome to JShell -- Version 12.0.2
|  For an introduction type: /help intro

jshell> import java.lang.invoke.*

jshell> MethodHandles.Lookup l = MethodHandles.lookup();
l ==>

jshell> l.in(void.class)
$3 ==> void/public

jshell> l.in(int.class)
$4 ==> int/public

jshell> l.in(int[].class)
$5 ==> [I/public
```
3.JDK13
```
# jdk_13.0.2_8_hotspot/bin/jshell -J-showversion
openjdk version "13.0.2" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.2+8)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.2+8, mixed mode, sharing)
|  Welcome to JShell -- Version 13.0.2
|  For an introduction type: /help intro

jshell> import java.lang.invoke.*

jshell> MethodHandles.Lookup l = MethodHandles.lookup();
l ==>

jshell> l.in(void.class)
$3 ==> void/public

jshell> l.in(int.class)
$4 ==> int/public

jshell> l.in(int[].class)
$5 ==> [I/public
```
4.JDK14
```
# jdk14_hotspot_04_14_2020/bin/jshell -J-showversion
java version "14.0.1" 2020-04-14
Java(TM) SE Runtime Environment (build 14.0.1+4)
Java HotSpot(TM) 64-Bit Server VM (build 14.0.1+4, mixed mode, sharing)
|  Welcome to JShell -- Version 14.0.1
|  For an introduction type: /help intro

jshell> import java.lang.invoke.*

jshell> MethodHandles.Lookup l = MethodHandles.lookup();
l ==>

jshell> l.in(void.class)
|  Exception java.lang.IllegalArgumentException: void is a primitive class
|        at MethodHandles$Lookup.in (MethodHandles.java:1476)
|        at (#3:1)

jshell> l.in(int.class)
|  Exception java.lang.IllegalArgumentException: int is a primitive class
|        at MethodHandles$Lookup.in (MethodHandles.java:1476)
|        at (#4:1)

jshell> l.in(int[].class)
|  Exception java.lang.IllegalArgumentException: class [I is an array class
|        at MethodHandles$Lookup.in (MethodHandles.java:1478)
|        at (#5:1)
```
Based on the results above, I will restore the previous changes back to only have the check enabled on JDK14 and removed the test cases that was updated for the type check.

Reply:
The validity check has only been enabled on JDK14 and the corresponding test case were also disabled on JDK14. Will launch personal builds to see whether it works correctly as expected.